public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js

```diff
@@ -1,29 +1,70 @@
-mciModule.controller('PerformanceDiscoveryCtrl', function($scope) {
+mciModule.controller('PerformanceDiscoveryCtrl', function(
+  $scope, $q, ApiTaskdata, ApiV1, PerfDiscoveryService, PERF_DISCOVERY
+) {
   var vm = this;
 
   vm.revisionSelect = {
-    options: [
-      {id: 1, name: '4ef5fx',},
-      {id: 2, name: 'v52fe3',},
-      {id: 3, name: 'd43ap0',},
-    ],
+    options: [],
+    selected: null,
   }
 
-  vm.revisionSelect.selected = vm.revisionSelect.options[0]
-
   vm.tagSelect = {
-    options: [
-      {id: 1, name: 'v3.6.1',},
-      {id: 2, name: 'v3.6.0',},
-      {id: 3, name: 'v3.5.2',},
-    ],
+    options: [],
+    selected: null,
   }
 
-  vm.tagSelect.selected = vm.tagSelect.options[0]
+  // TODO use globally defined project id
+  var projectId = 'sys-perf'
+
+  ApiV1.getProjectVersions(projectId).then(function(data) {
+    vm.versions = data.data.versions
+    vm.revisionSelect.options = _.map(data.data.versions, function(d, i) {
+      return {id: i, name: d.revision}
+    })
+    vm.revisionSelect.selected = vm.revisionSelect.options[0]
+
+    vm.tagSelect.options = _.map(data.data.versions, function(d, i) {
+      return {id: i, name: d.revision}
+    })
+    // Choose the second item, if exists, or the first if no
+    vm.tagSelect.selected = vm.tagSelect.options[
+      _.min([1, vm.tagSelect.options.length - 1])
+    ]
+    vm.changeRevision(data.data.versions[0].revision)
+  })
+
+  vm.changeRevision = function(revision) {
+    version = _.findWhere(vm.versions, {
+      revision: revision
+    })
+
+    var buildTaskData = _.reduce(version.builds, function(items, build) {
+      var taskItems = _.map(build.tasks, function(v, k) {
+        return {
+          build: build.name,
+          task: k,
+          task_id: v.task_id,
+        }
+      })
+      return items.concat(taskItems)
+    }, [])
+
+    var ctx = PerfDiscoveryService.extractTasks(version)
+
+    PerfDiscoveryService.getRows(
+      PerfDiscoveryService.processData(
+        PerfDiscoveryService.queryData(
```

Comment:
This comment is probably for the processData function in the service, but it should more diligently handle when the data returned by the request is null. There are versions where a given task will not run and thus will not have anything in the json document (and I'm not sure that there's a way to know this ahead of time).

Here's the error that I see in the console:
angular.js:14700 TypeError: Cannot read property 'current' of null
    at PerfDiscoveryService.js?hash=8b31bfb6dc7e92be8222b435a1e739ee:133
    at Function.m.each.m.forEach (underscore-min.js:5)
    at PerfDiscoveryService.js?hash=8b31bfb6dc7e92be8222b435a1e739ee:132
    at processQueue (angular.js:17051)
    at angular.js:17095
    at Scope.$digest (angular.js:18233)
    at Scope.$apply (angular.js:18531)
    at done (angular.js:12547)
    at completeRequest (angular.js:12785)
    at XMLHttpRequest.requestLoaded (angular.js:12702) "Possibly unhandled rejection: {}"

Reply:
I covered some edge case. Missed this one.